sympy/core/power.py

```diff
@@ -520,7 +520,7 @@ def _check(ct1, ct2, old):
         if old == self.base:
             return new**self.exp._subs(old, new)
 
-        if old.func is self.func and self.base is old.base:
+        if old.func is self.func and self.base == old.base:
```

Comment:
Yes, good catch. During consolidation of the code I only had in mind the atomic base (x, E, 2, etc...).